Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -167,38 +154,26 @@ extension Message {
   ///     `BinaryEncodingError.missingRequiredFields`.
   ///   - options: The BinaryDecodingOptions to use.
   /// - Throws: `BinaryDecodingError` if decoding fails.
-  @inlinable
```

Comment:
I went back to read @Lukasa 's great writeup in the comments of #914 in hopes of answering a question for myself, but I failed to find an answer. So, here goes: what changed such that we no longer need/want this generic method to be `@inlinable`, and are we giving up anything? Like, did generic non-`@inlinable` methods get better in 5.0+, or is the win small enough that we don't care?

I thought the value in `_merge` was not its reuse (which is gone) but rather the way it permitted calling modules to merely give us an `UnsafeRawBufferPointer` on their own terms.

Reply:
I don't think anything has meaningfully changed to enable this. Swift's cross-module optimisation functionality _should_ be able to make this `@inlinable` unnecessary, but that is not on by default and somewhat painful to enable. So I think the thunk to `_merge` and the `@inlinable` remains useful.

Reply:
Before there were <5.0 code paths that also called `_merge`, but once I removed all the `#if` checks we ended up with only this one spot calling `_merge`, so that's why I folded it together.  So I thought the only net change here was that the `try bytes.withUnsafeBytes()` frame would be the only net shift, and that likely would matter from a performance POV.  But rereading #194, I guess the issue is if the calling module has custom `ContiguousBytes` with `@inlinable` support for the `withUnsafeBytes` call?

Reply:
Specifically it's if this is called with any type _other_ than `Data`, in which case this code path will be hit and we can inline the call to `withUnsafeBytes`.

Reply:
Updated with a code reference to the original thread.